conseil-api-tests/src/main/scala/tech/cyptonomic/conseil/api/tests/ApiTestRun.scala

```diff
@@ -0,0 +1,289 @@
+          |     "aggregation": [],
+          |     "limit": 10
+          |}
+          |""".stripMargin.replace("%FIELDS%", entity._2.asJson.toString)
+
+      println("\n\n\nTesting Tezos " + entity._1 + " Query\n")
+      val result: String = sendConseilQuery(httpClient, CONSEIL.withPath(Requests.getTezosQueryPath(entity._1)), queryString)
+      println(result)
+
+      entity._2.foreach(attribute => {
+        validateJsonByKey(attribute, result)
+      })
+
+    })
+
+  }
+
+  /**
+   * Sends a query to make sure each attribute data is valid
+   * @param httpClient The BlazeClientBuilder[IO] object that sends the conseil requests
+   * @param entityAttributes A Map that has entities as keys, and an array of attributes as its values
+   */
+  def validateAttributeData(httpClient: BlazeClientBuilder[IO], entityAttributes: Map[String, Array[String]] = Requests.TEZOS_ENTITY_ATTRIBUTES): Unit = {
+
+    entityAttributes.foreach(entity => {
+
+      entity._2.foreach(attribute => {
+
+        val queryString: String =
+          """
+            |{
+            |     "field": ["%FIELD%"],
+            |     "predicates": [
+            |         {
+            |             "field": "%FIELD%",
+            |             "operation": "isnull",
+            |             "set": [""],
+            |             "inverse": true
+            |         }
+            |     ],
+            |     "orderBy": [],
+            |     "aggregation": [],
+            |     "limit": 1
+            |}
+            |""".stripMargin.replace("%FIELD%", attribute)
+
+        println("\n\n\nValidating Tezos " + entity._1 + " " + attribute + " Data\n")
+        var result: String = ""
+        try {
+          result = sendConseilQuery(httpClient, CONSEIL.withPath(Requests.getTezosQueryPath(entity._1)), queryString)
+        } catch {
+          case x: org.http4s.client.UnexpectedStatus => {
+
+            val queryString: String =
+              """
+                |{
+                |     "field": ["%FIELD%"],
+                |     "predicates": [],
+                |     "orderBy": [],
+                |     "aggregation": [],
+                |     "limit": 1
+                |}
+                |""".stripMargin.replace("%FIELD%", attribute)
+
+            result = sendConseilQuery(httpClient, CONSEIL.withPath(Requests.getTezosQueryPath(entity._1)), queryString)
+
+          }
+        }
+
+        println(result)
+
+        validateJsonByKey(attribute, result, notNull = true)
+
+      })
+    })
+  }
+
+  /**
+   * Validate a query output by searching for JSON keys
+   * @param key The json key to search for
+   * @param result The result of a query to search in
+   * @param notNull Whether the key's value should be allowed to be null or not
+   */
+  def validateJsonByKey(key: String, result: String, notNull: Boolean = false): Unit = {
+
+    if(result.compareTo("") == 0) throw new NullPointerException("Query Returned an Empty String")
```

Comment:
why not just `result == ""`?

Reply:
Oh I thought like Java, string comparison might only be properly done through comparison methods